Check Pronunciation: NullPointerException after Microphone Permissions

https://couchdb.ankidroid.org/acralyzer/_design/acralyzer/index.html#/report-details/b5e73b36-aaf9-462b-8ac0-fa448202323c

2.12alpha13

```
1	06-18 22:06:30.471 I/AnkiDroid( 2100): Applying background
2	06-18 22:06:30.692 I/AnkiDroid( 2100): colOpen: true
3	06-18 22:06:30.697 I/AnkiDroid( 2100): No startup screens required
4	06-18 22:06:30.703 I/AnkiDroid( 2100): AnkiActivity::onStart
5	06-18 22:06:30.748 I/AnkiDroid( 2100): AnkiActivity::onResume
6	06-18 22:06:33.049 I/AnkiDroid( 2100): AnkiActivity::onPause
7	06-18 22:06:33.202 I/AnkiDroid( 2100): AnkiActivity::onStop
8	06-18 22:06:37.101 I/AnkiDroid( 2100): AnkiActivity::onStart
9	06-18 22:06:37.147 I/AnkiDroid( 2100): AnkiActivity::onResume
10	06-18 22:06:41.954 I/AnkiDroid( 2100): Updating deck list UI
11	06-18 22:06:54.634 I/AnkiDroid( 2100): DeckPicker:: Selected deck with id 1581824097593
12	06-18 22:06:54.676 I/AnkiDroid( 2100): AnkiActivity::onPause
13	06-18 22:06:54.849 I/AnkiDroid( 2100): AnkiActivity::onCreate
14	06-18 22:06:55.001 I/AnkiDroid( 2100): Obtaining card
15	06-18 22:06:55.003 I/AnkiDroid( 2100): AnkiActivity::onStart
16	06-18 22:06:55.039 I/AnkiDroid( 2100): AnkiActivity::onResume
17	06-18 22:06:55.528 I/AnkiDroid( 2100): AbstractFlashcardViewer:: Question successfully shown for card id 1487304721486
18	06-18 22:06:55.747 I/AnkiDroid( 2100): AnkiActivity::onStop
19	06-18 22:06:55.749 I/AnkiDroid( 2100): flush - Saving information to DB...
20	06-18 22:06:57.559 I/AnkiDroid( 2100): Play all: Completed - releasing sound
21	06-18 22:06:59.400 I/AnkiDroid( 2100): Reviewer:: Record mic
22	06-18 22:06:59.466 I/AnkiDroid( 2100): AnkiActivity::onPause
```

```
java.lang.RuntimeException: Failure delivering result ResultInfo{who=@android:requestPermissions:, request=0, result=-1, data=Intent { act=android.content.pm.action.REQUEST_PERMISSIONS (has extras) }} to activity {com.ichi2.anki/com.ichi2.anki.Reviewer}: java.lang.NullPointerException: Attempt to invoke virtual method 'void android.widget.LinearLayout.addView(android.view.View)' on a null object reference
at android.app.ActivityThread.deliverResults(ActivityThread.java:4126)
at android.app.ActivityThread.handleSendResult(ActivityThread.java:4169)
at android.app.ActivityThread.-wrap20(ActivityThread.java)
at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1552)
at android.os.Handler.dispatchMessage(Handler.java:102)
at android.os.Looper.loop(Looper.java:154)
at android.app.ActivityThread.main(ActivityThread.java:6186)
at java.lang.reflect.Method.invoke(Native Method)
at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:889)
at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:779)
at de.robv.android.xposed.XposedBridge.main(XposedBridge.java:107)
Caused by: java.lang.NullPointerException: Attempt to invoke virtual method 'void android.widget.LinearLayout.addView(android.view.View)' on a null object reference
at com.ichi2.anki.Reviewer.toggleMicToolBar(Reviewer.java:11)
at com.ichi2.anki.Reviewer.onRequestPermissionsResult(Reviewer.java:2)
at android.app.Activity.dispatchRequestPermissionsResult(Activity.java:7089)
at android.app.Activity.dispatchActivityResult(Activity.java:6941)
at android.app.ActivityThread.deliverResults(ActivityThread.java:4122)
... 10 more
```

Hello,

I have checkout the 2.12alpha13 build and run on my HUAWEI P10Lite and I was not able to reproduce.
Here is what I have tested:
- start app
- Grant media access
- open a deck to review
- activate the pronunciation checker
- Refuse the microphone access
-> The mic tool bar is not displayed, no crash
- activate the pronunciation checker
- Grand mic access
-> The mic tool bar is displayed, no crash 

I am not familiar with https://couchdb.ankidroid.org/acralyzer/_design/acralyzer/index.html#/report-details/b5e73b36-aaf9-462b-8ac0-fa448202323c
What can I do with it, I follow the link, but it does not display anything. 
I do not find how to create an account

Then reading the following trace log:

`Caused by: java.lang.NullPointerException: Attempt to invoke virtual method 'void android.widget.LinearLayout.addView(android.view.View)' on a null object reference
at com.ichi2.anki.Reviewer.toggleMicToolBar(Reviewer.java:11)`

The only null object could be micToolBarLayer which is statically defined.
It looks like if `LinearLayout micToolBarLayer = findViewById(R.id.mic_tool_bar_layer);` where not found.
How could it be possible ?

Do you have some reproduction context to help me reproduce ?

--- a/AnkiDroid/src/main/res/layout/reviewer_fullscreen.xml
+++ b/AnkiDroid/src/main/res/layout/reviewer_fullscreen.xml
@@ -17,9 +17,15 @@
                      android:layout_width="fill_parent"
                      android:layout_height="wrap_content"
                      android:layout_alignParentTop="true"/>
+            <include
+                layout="@layout/reviewer_mic_tool_bar"
+                android:layout_width="match_parent"
+                android:layout_height="wrap_content"
+                android:layout_below="@id/top_bar" />
             <include layout="@layout/reviewer_flashcard_fullscreen"
                      android:layout_width="fill_parent"
-                     android:layout_height="fill_parent"/>
+                     android:layout_height="fill_parent"
+                     android:layout_below="@id/mic_tool_bar_layer"/>
             <include layout="@layout/reviewer_answer_buttons"/>
         </RelativeLayout>
         <!-- Controls that are overlaid over the main content when the user interrupts immersive mode -->

--- a/AnkiDroid/src/main/res/layout/reviewer_fullscreen_noanswers.xml
+++ b/AnkiDroid/src/main/res/layout/reviewer_fullscreen_noanswers.xml
@@ -13,7 +13,12 @@
             android:layout_width="match_parent"
             android:layout_height="match_parent">
             <!-- Main content that takes up the fullscreen -->
-            <include layout="@layout/reviewer_flashcard_fullscreen_noanswers"/>
+            <include
+                layout="@layout/reviewer_mic_tool_bar"
+                android:layout_width="match_parent"
+                android:layout_height="wrap_content" />
+
+            <include layout="@layout/reviewer_flashcard_fullscreen_noanswers" />
             <!-- Controls that are overlaid over the main content when the user interrupts immersive mode -->
             <FrameLayout
                 android:layout_width="match_parent"

